spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala

```diff
@@ -36,8 +36,16 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
  */
 @SerialVersionUID(- 8525406230282608924L)
-class Input[T: ClassTag]()(implicit ev: TensorNumeric[T])
+class Input[T: ClassTag](val inputShape: Array[Int])(implicit ev: TensorNumeric[T])
   extends AbstractModule[Activity, Activity, T] {
+
+  if (inputShape != null) {
+    val batchInputShape = Array(-1) ++ inputShape
```

Comment:
what if the passed in inputShape has batch dimension...

Reply:
After discuss with @jason-dai ,  the computing logic would only take care of the shape a single record without batch.